docs/template_user_configuration_file.yaml

```diff
@@ -0,0 +1,22 @@
+InternalTransferReferences:
```

Comment:
### This comment applies to the whole file.

I think we can have examples that really communicate real world things. For `ExpenseCategoryDefinition` and `IncomeCategoryDefinition` examples, have a look at some of the mocked/test configs in unit tests.

A simple thing like `#pub` or `mac donalds` clarifies more than just repeating variable names.